src/pudl/extract/eia860.py

```diff
@@ -40,6 +50,7 @@ def process_raw(self, df, page, **partition):
         if page in pages_eia860m:
             df = df.assign(data_source='eia860')
             self.cols_added.append('data_source')
+        df = fix_leading_zero_gen_ids(df)
```

Comment:
Should at least get into the standard `process_raw` method in the `GenericExtractor`. Even though it is basically not being run right now because it is always overwritten? At least as a template to let our future selves know it should be added? Or is it specific enough to tables which contain generator ids that it should be done for all datasets... obviously it would just do nothing for non-gen tables but we would probably remove it entirely for a non-gen data set... hm. just musing on your original point of maybe wanting this in a standard place.

In truth it feels more like it should be an early phase transform step. In which case it could be applied to all dfs within the data set specific main transform function.

Reply:
I totally agree that this should be a pre-transform step, not an extract step. My first attempt was to integrate it there, but the differences between how the eia860, eia923, and eia861 transform modules are structured meant that it wasn't straightforward and I broke everything. Integrating it upstream was easier for now. But as we standardize those modules, probably with a common class definition that they all use, I totally agree.

I think this is an EIA specific early transform fix though, not a generic one, though we could make it into something more generic that can be applied to any "string" field that contains a subset of integer fields, which should ditch any spurious leading zeroes. But in that case we would need to tell it which columns it should apply to, or have that information stored in the metadata about the columns somehow. Unfortunately `generator_id` is a little too generic a name (I think we should go to `generator_id_eia` personally).